packages/zquery/src/index.test.ts

```diff
@@ -1,57 +1,222 @@
 import { beforeEach, describe, expect, it, vi } from 'vitest';
 import { createZQuery } from '.';
-import { UseStore } from './types';
+import { MOCK_PUPPY_PHOTOS, PuppyPhoto, State } from './test/mock-state';
+import { StoreApi, UseBoundStore, create } from 'zustand';
+import { ZQuery } from './types';
+import { waitFor } from '@testing-library/react';
 
 describe('createZQuery()', () => {
-  type MockState = null;
-  let mockUseStore: UseStore<MockState>;
-  let mockState: MockState;
+  let zQuery: ZQuery<'puppyPhotos', PuppyPhoto[], []>;
```

Comment:
I've just tweaked the setup for the tests, but the first few tests are the same until down where I've commented below.